java/src/test/java/com/logicalclocks/hsfs/TestHopsworksExternalClient.java

```diff
@@ -53,6 +53,21 @@
         .willReturn(success().body(HttpBodyConverter.json(credentials)))
     ));
 
+  @Test
+  public void testReadAPIKey() throws IOException, FeatureStoreException {
+    CloseableHttpClient httpClient = HttpClients.createSystem();
+    try {
+      HopsworksConnection hc = HopsworksConnection.builder().host("35.241.253.100").hostnameVerification(false)
+              .project("demo_featurestore_admin000")
+              .apiKeyValue("ovVQksgJezSckjyK.ftO2YywCI6gZp4btlvWRnSDjSgyAQgCTRAoQTTSXBxPRMo0Dq029eAf3HVq3I6JO").build();
+      System.out.println("Connected");
+      FeatureStore fs = hc.getFeatureStore();
+      Assert.assertTrue(fs != null);
```

Comment:
I am not sure `hc.getFeatureStore()` will ever return `null` but will rather throw an exception if it can't do the rest call. So I don't think this test will ever fail with an AssertionError and hence always run into the catch clause.

Reply:
I think it is an IOException that it is throwing. Fabio wants to change it to a FeatureStoreException

Reply:
The code is removed.